frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderButtonRow.tsx

```diff
@@ -0,0 +1,315 @@
+        isEditing,
+        isFullscreen,
+      }),
+  },
+  [BUTTON_CONFIG.DASHBOARD_EMBED_ACTION]: {
+    component: () => <DashboardEmbedAction />,
+    enabled: ({ dashboard, isPublic, isEmpty }) =>
+      !isEmpty && !isPublic && dashboard && !dashboard?.archived,
+  },
+  [BUTTON_CONFIG.REFRESH_WIDGET]: {
+    component: ({
+      refreshPeriod,
+      setRefreshElapsedHook,
+      onRefreshPeriodChange,
+    }) => (
+      <RefreshWidgetButton
+        period={refreshPeriod}
+        setRefreshElapsedHook={setRefreshElapsedHook}
+        onChangePeriod={onRefreshPeriodChange}
+      />
+    ),
+    enabled: ({ dashboard, isEmpty }) => !isEmpty && !dashboard?.archived,
+  },
+  [BUTTON_CONFIG.NIGHT_MODE_TOGGLE]: {
+    component: ({ isNightMode, onNightModeChange }) => (
+      <NightModeToggleButton
+        isNightMode={isNightMode}
+        onNightModeChange={onNightModeChange}
+      />
+    ),
+    enabled: ({
+      isFullscreen,
+      dashboard,
+      hasNightModeToggle,
+      onNightModeChange,
+    }) =>
+      Boolean(
+        isFullscreen &&
+          !dashboard.archived &&
+          hasNightModeToggle &&
+          !!onNightModeChange,
+      ),
+  },
+  [BUTTON_CONFIG.FULLSCREEN_TOGGLE]: {
+    component: ({ isFullscreen, onFullscreenChange }) => (
+      <FullscreenToggle
+        isFullscreen={isFullscreen}
+        onFullscreenChange={onFullscreenChange}
+      />
+    ),
+    enabled: ({ isFullscreen, isEmpty, isPublic }) =>
+      !isEmpty && (isPublic || isFullscreen),
+  },
+  [BUTTON_CONFIG.DASHBOARD_HEADER_ACTION_DIVIDER]: {
+    component: () => <DashboardHeaderActionDivider />,
+    enabled: ({ isEditing }) => !isEditing,
+  },
+  [BUTTON_CONFIG.DASHBOARD_BOOKMARK]: {
+    component: () => <DashboardBookmark />,
+    enabled: ({ isEditing, dashboard }) => !isEditing && !dashboard.archived,
+  },
+  [BUTTON_CONFIG.DASHBOARD_INFO]: {
+    component: () => <DashboardInfoButton />,
+    enabled: ({ isEditing }) => !isEditing,
+  },
+  [BUTTON_CONFIG.DASHBOARD_ACTION_MENU]: {
+    component: ({
+      onFullscreenChange,
+      isFullscreen,
+      dashboard,
+      canEdit,
+      location,
+    }) => (
+      <DashboardActionMenu
+        items={getExtraButtons({
+          onFullscreenChange,
+          isFullscreen,
+          dashboard,
+          canEdit,
+          pathname: location.pathname,
+        })}
+      />
+    ),
+    enabled: ({ isFullscreen, isEditing, isAnalyticsDashboard, dashboard }) =>
+      !isFullscreen &&
+      !isEditing &&
+      !isAnalyticsDashboard &&
+      !dashboard.archived,
+  },
+  [BUTTON_CONFIG.FULLSCREEN_ANALYTICS_DASHBOARD]: {
+    component: ({ isFullscreen, onFullscreenChange }) => (
+      <FullscreenAnalyticsDashboard
+        isFullscreen={isFullscreen}
+        onFullscreenChange={onFullscreenChange}
+      />
+    ),
+    enabled: ({ isAnalyticsDashboard }) => isAnalyticsDashboard,
+  },
+};
+
+export const DashboardHeaderButtonRow = withRouter(
+  ({
```

Comment:
You should add a function name here so it's shown up in React dev tools. Otherwise, you couldn't search for `DashboardHeaderButtonRow` in the React dev tool.

```suggestion
  function DashboardHeaderButtonRow({
```

Reply:
I actually ended up moving `withRouter` to the `DASHBOARD_ACTION_MENU` - it's the only one that uses the router so no point in passing the location to any of the other elements. I hope we can get to a point where we don't need to pass props at all in the future.